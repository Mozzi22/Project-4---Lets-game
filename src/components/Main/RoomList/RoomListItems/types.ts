import { EnumType } from "typescript"

export type TRegistrationValue = {
     room_id: string, room_name?: string 
}
export type TRegistrationData = {
    name: string, value: TRegistrationValue
}
export interface IRoomListItems {
    id: string;
    img?: string;
    color?: string;
    login?: string;
    setValue: (arg: TRegistrationData) => void;
    currentRoomName: string;
    rooms: string;
    typeGame: string;
}
export interface IStRoom {
    bgColor?: string;
}
