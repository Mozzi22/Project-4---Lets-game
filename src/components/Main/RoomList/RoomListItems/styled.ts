import styled from 'styled-components';
import { bgColorDefault, colorDefault } from 'src/components/UI/baseLayout';
import { IStRoom } from './types';

export const StRoomListItems = styled.div`
    display: flex;
    overflow: hidden;
    min-height: 60px;
    align-items: center;
    color: ${({ color = colorDefault }) => color};
    margin: 0 20px 20px;
    border-radius: 5px;
    padding: 7px 15px;
    background-color: #2acbb9;
    cursor: pointer;
        ul {
            padding-left: 1rem;
            list-style-type: none;
        }
        :hover {
            color: ${({ color = bgColorDefault }) => color};
            span{
                color: ${({ color = colorDefault }) => color};
            }
        }
`;
export const StRoom = styled.div < IStRoom > `
    display:flex;
    justify-content: flex-start;
    align-items: center;
    width: 100%;
    max-height: 55px;
    overflow: hidden;
        h2 {
            font-size: 18px;
            margin: 0;
            overflow-wrap: break-word;
        }
        h3 {
            font-size: 16px;
            text-transform: capitalize;
            margin: 0;
            overflow-wrap: break-word;
            line-height: 30px;
        }
        div {
            width: 80%;
            color: #eee;
            max-height: 60px;
            padding: 20px 0;
        }
`;
