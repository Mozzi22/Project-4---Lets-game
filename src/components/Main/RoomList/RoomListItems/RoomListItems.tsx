import React, { useEffect, useState } from 'react';
import { bgColorDefault } from 'src/components/UI/baseLayout';
import Button from 'src/components/UI/Button';
import { useTranslation } from 'react-i18next';
import { StRoomListItems, StPhoto, StRoom } from './styled';
import { IRoomListItems } from './types';

const RoomListItems: React.FC<IRoomListItems> = ({
  id,
  img,
  color,
  content,
  setValue,
  currentRoomName,
}) => {
    const { t } = useTranslation();
    const handleJoinToGame = () => {
    };
  const handleClick = () => {
    setValue({ name: 'currentRoom', value: { room_id: id, room_name: content } });
  };
  useEffect(() => {
    if (currentRoomName === content) {
      return setState({ ...state, selected: true });
    }
    setState({ ...state, selected: false });
  }, [currentRoomName]);
  const [state, setState] = useState({
    error: false,
    selected: false,
  });
  const onError = () => setState({ ...state, error: true });
  return (
    <StRoomListItems color={state.selected ? bgColorDefault : color} onClick={handleClick}>
      <StRoom>
        <p>{content}</p>
      </StRoom>
      <Button
            id='join_to_game'
            name='join_to_game'
            onClick={handleJoinToGame}
            content={t('join_to_game')}
            width="200px"
            margin="15px 0"
      />
    </StRoomListItems>
  );
};

export default RoomListItems;
