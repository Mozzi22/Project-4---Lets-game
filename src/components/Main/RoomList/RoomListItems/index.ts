import RoomListItems from './RoomListItems';

export default RoomListItems;
