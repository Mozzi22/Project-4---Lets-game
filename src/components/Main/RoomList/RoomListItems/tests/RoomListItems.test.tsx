import React from 'react';
import RoomListItems from '../RoomListItems';
import { mount, shallow } from 'enzyme';

describe('SearchNoRes', () => {
    it('Should match snapshot', () => {
        const component = shallow(<RoomListItems/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StRoomListItems', () => {
        const component = mount(<RoomListItems/>);
        expect(component.find('styled__StRoomListItems')).toHaveLength(1);
    });
    it('should render StRoom', () => {
        const component = mount(<RoomListItems/>);
        expect(component.find('styled__StRoom')).toHaveLength(1);
    });
    it('should render StNoResText', () => {
        const component = mount(<RoomListItems/>);
        expect(component.find('Button')).toHaveLength(1);
    });
});