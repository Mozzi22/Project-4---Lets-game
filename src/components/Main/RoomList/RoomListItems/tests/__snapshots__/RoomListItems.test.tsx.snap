// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchNoRes Should match snapshot 1`] = `"<div class=\\"styled__StRoomListItems-sc-19jz83q-0 gGwSht\\"><div class=\\"styled__StRoom-sc-19jz83q-2 jTeqUu\\"><p></p></div><button id=\\"join_to_game\\" name=\\"join_to_game\\" type=\\"button\\" width=\\"200px\\" class=\\"styled__StButton-sc-15f3n0p-0 fpLSlp\\">join_to_game</button></div>"`;
